src/stores/room-list/algorithms/Algorithm.ts

```diff
@@ -237,6 +242,10 @@ export class Algorithm extends EventEmitter {
         // Lie to the algorithm and remove the room from it's field of view
         this.handleRoomUpdate(val, RoomUpdateCause.RoomRemoved);
 
+        // handleRoomUpdate may have modified this._stickyRoom. Convince the
+        // compiler of this fact.
+        this._stickyRoom = this.stickyRoomMightBeModified();
```

Comment:
Help welcome on how to do this better. I didn't want to just suppress the error on `if (this._stickyRoom.room !== val)`. The compiler thinks `this._stickyRoom` is `never` in there because `this._stickRoom` was set to `null` above. It is missing the fact that it might have been modifed inside `handleRoomUpdate`.

Reply:
Yeah, I've had this before and it seems like something typescript ought to be smarter about. I think I had cast it to `type | null` but this seems fine too.